dtfs-central-api/src/utils/report-period.ts

```diff
@@ -27,3 +29,40 @@ export const getPreviousReportPeriodStart = ({ month, year }: MonthAndYear): Mon
 
 export const isEqualReportPeriodStart = (reportPeriodStart1: MonthAndYear, reportPeriodStart2: MonthAndYear): boolean =>
   reportPeriodStart1.year === reportPeriodStart2.year && reportPeriodStart1.month === reportPeriodStart2.month;
+
+export const getCurrentReportPeriodForBankSchedule = (bankReportPeriodSchedule: BankReportPeriodSchedule): ReportPeriod => {
+  const previousMonthDate = subMonths(new Date(), 1);
+  const targetMonth = getOneIndexedMonth(previousMonthDate);
+  const targetYear = previousMonthDate.getFullYear();
+
+  const scheduleInCurrentYear = bankReportPeriodSchedule.find((schedule) => targetMonth >= schedule.startMonth && targetMonth <= schedule.endMonth);
+  if (scheduleInCurrentYear) {
+    return {
+      start: {
+        month: scheduleInCurrentYear.startMonth,
+        year: targetYear,
+      },
+      end: {
+        month: scheduleInCurrentYear.endMonth,
+        year: targetYear,
+      },
+    };
+  }
+
+  const scheduleOverlappingYear = bankReportPeriodSchedule.find((schedule) => schedule.startMonth > schedule.endMonth);
+  if (!scheduleOverlappingYear) {
+    throw new Error('Failed to get a report period');
```

Comment:
Do we need `console.error` or will the calling function `catch()` block have a `console.error`?